flyteplugins/go/tasks/pluginmachinery/flytek8s/k8s_resource_adds.go

```diff
@@ -127,6 +127,24 @@ func DecorateEnvVars(ctx context.Context, envVars []v1.EnvVar, taskEnvironmentVa
 		value := os.Getenv(envVarName)
 		envVars = append(envVars, v1.EnvVar{Name: k, Value: value})
 	}
+	for _, configMapName := range config.GetK8sPluginConfig().DefaultEnvVarsFromConfigMaps {
+		envVars = append(envVars, v1.EnvVar{
+			ValueFrom: &v1.EnvVarSource{
+				ConfigMapKeyRef: &v1.ConfigMapKeySelector{
+					Key: configMapName,
+				},
+			},
+		})
+	}
```

Comment:
actually, I gave this a try and the code is not quite correct. In order to load config maps as environment variables you need to specify the keys you're going to load explicitly.

So the config file would look like:
```
default-env-vars-from-configmaps:
  test-config-map:
    - key1
    - key2
  config-map-2:
    - another-key-1
    - another-key-2
```

Which would cause the 4 env vars `key1`, `key2`, `another-key-1`, and `another-key-2` in the pod.

This has the added benefit that you can guarantee that the names do not conflict by visually inspecting the config file (and we could even add some validation there afterwards).

In order to load the config map we need to modify the type of `DefaultEnvVarsFromConfigMaps` to be an dict of list, such that we can do something like:
```
for configMapName, keys := range config.GetK8sPluginConfig().DefaultEnvVarsFromConfigMaps {
    for key := range keys {
        envVars = append(envVars, v1.EnvVar{
            Name: key,
            ValueFrom: &v1.EnvVarSource{
                ConfigMapKeyRef: &v1.ConfigMapKeySelector{
                    Key: key,
                    LocalObjectReference: v1.LocalObjectReference{
                        Name: configMapName,
                    },
                },
            },
        })
    }
}
```

We might need to revisit the namespacing of the env vars (maybe use the config map name as the prefix?).

I didn't investigate the support for secrets, but I'm assuming that it should be pretty similar.

Reply:
Interesting. It's supported in that manner via raw k8s, so I was assuming we'd be able to: https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#configure-all-key-value-pairs-in-a-configmap-as-container-environment-variables

Reply:
That's definitely interesting. The use is slightly different, right? In the case of https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#configure-all-key-value-pairs-in-a-configmap-as-container-environment-variables we have to fill in the `envFrom` field in the container, whereas the support we have today is for the `env` field. Using `envFrom` should be pretty simple though.

The docs don't mention, but I'm not sure if we can the mix `envFrom` and `env`. What happens if there's a clash (i.e. two config maps that define the same key)?

Reply:
Yeah, envFrom should be the target to the container; I'll see what I can do to dig through this deeper.

For the conflicts, `env` will override anything else specified, so the ultimate priority would be: `env` > `envFrom` > `Dockerfile` builtin ENV for priorities

Reply:
@eapolinario I've pushed up a change that allows direct `envFrom[]` into k8s pods as originally intended, but I see your point on pulling specific keys on configMaps/secrets into the env array using the pick-and-choose method.

If you're ok with it I've renamed the config keys to be more specific that this is for the envFrom section in k8s to leave room for future `env[].valueFrom.configMapRef` / `env[].valueFrom.secretRef` to be implemented as you described